hr_employee_calendar_planning/models/hr_employee.py

```diff
@@ -185,7 +185,8 @@ def create(self, vals_list):
             and res.filtered(lambda x: not x.calendar_ids)
         ):
             raise UserError(_("You can not create employees without any calendar."))
-        res.filtered("calendar_ids").regenerate_calendar()
+        for employee in res.filtered("calendar_ids"):
+            employee.sudo().regenerate_calendar()
```

Comment:
Why changing this to one record calls?

Reply:
@pedrobaeza as you see in my PR description this change is to avoid the singletone error. If you find another better way just let me know

About the test changes I also made a comment on the description of the work done: "On tests assert on number of attendances changed slightly because of the lunch break"

Let me know if you need further explanations